scrapy/extensions/closespider.py

```diff
@@ -74,9 +91,16 @@ def error_count(self, failure: Failure, response: Response, spider: Spider) -> N
 
     def page_count(self, response: Response, request: Request, spider: Spider) -> None:
         self.counter["pagecount"] += 1
+        self.counter["pagecount_since_last_item"] += 1
         if self.counter["pagecount"] == self.close_on["pagecount"]:
             assert self.crawler.engine
             self.crawler.engine.close_spider(spider, "closespider_pagecount")
```

Comment:
Maybe add an early return at the end of this `if`. No need to check if the condition below has been met if this one has.